### DISCUSSION TITLE
Regex to obfuscate certain outputs in recording

### DISCUSSION THREAD

**Discussion by ArcAster** [asciinema/discussions#60] [Q&A]
For instance, if I want to share a workflow using nmap or some other tool that may expose sensitive details or credentials in the recorded terminal output, it would be incredibly convenient to be able to apply a regex match and substitute the match with null or pre-defined characters.

An example I've run into is wanting to match all ip addresses and replace them with 000.000.0.0 or regex match a user or log id.

The inability to sanitize asciinema output has been a reason we haven't been using asciinema to create guides / tutorials at work since we have to have at least basic guarantees that credentials or sensitive info isn't ending up in the recordings.  (I'm aware that it is potentially possible to host your own asciinema recordings?)

**Comment by ku1ik** [comment]
What you can do is to record to local file (`asciinema rec demo.cast`), then run aws/sed/etc over this file to replace sensitive content. The recording is simple json-lines file (https://github.com/asciinema/asciinema/blob/develop/doc/asciicast-v2.md) so it shouldn't be hard to do it. Then you can review if nothing leaks, and upload the file with `asciinema upload demo.cast`.
